jupyterhub/handlers/base.py

```diff
@@ -978,14 +979,21 @@ def _fail_api_request(self, user):
                 port = 443 if host_info.scheme == 'https' else 80
             if port != Server.from_url(self.proxy.public_url).connect_port and port == self.hub.connect_port:
                 self.log.warning("""
-                    Detected possible direct connection to Hub's private ip: %s, bypassing proxy.
-                    This will result in a redirect loop.
-                    Make sure to connect to the proxied public URL %s
-                    """, self.request.full_url(), self.proxy.public_url)
-
+                Detected possible direct connection to Hub's private ip: %s, bypassing proxy.
+                This will result in a redirect loop.
+                 Make sure to connect to the proxied public URL %s
+                """, self.request.full_url(), self.proxy.public_url)
+                    
             # logged in as valid user, check for pending spawn
-            spawner = user.spawner
-
+            if server_name is None:
+                spawner = user.spawner
+            else:
+                spawner = user.spawners[server_name]
+                #if self.allow_named_servers:
```

Comment:
This `allow_named_servers` check should still be here

Reply:
I agree it should still be there. When I added it, it caused the tests to fail.
 
(The first time I added it, the tests failed due to a 400 HTTP Error being raised. When I deliberately changed HTTP error on line 995 to be a 500 HTTP error, the tests failed again, but this time due to a 500 HTTP error. So I am fairly confident, albeit not certain, that it was this block that caused the tests to fail.)

The test failure was something about /user/river or something like that. (I want to say `test_auth.py`, but my memory isn't fresh.) 

My guess would have been that the tests wouldn't use named servers for the UI, so I was somewhat confused why this section of the code was reached in the first place.

Anyway, I figured you wouldn't look at a failing PR, so I commented that block out.